Monitoring 4.19 Release Notes (#94163)
openshift/openshift-docs: release_notes/ocp-4-19-release-notes.adoc

[id="ocp-release-notes-monitoring_{context}"]
=== Monitoring

The in-cluster monitoring stack for this release includes the following new and modified features:

[id="ocp-4-19-monitoring-updates-to-monitoring-stack-components-and-dependencies"]
==== Updates to monitoring stack components and dependencies

This release includes the following version updates for in-cluster monitoring stack components and dependencies:

> * Alertmanager	to 0.28.1

Reviewer: ```suggestion
* Alertmanager to 0.28.1
```

* Prometheus to 3.2.1
* Prometheus Operator to 0.81.0
* Thanos to 0.37.2
* kube-state-metrics to 2.15.0
* node-exporter to 1.9.1

[id="ocp-4-19-monitoring-changes-to-alerting-rules"]
==== Changes to alerting rules

[NOTE]
====
Red{nbsp}Hat does not guarantee backward compatibility for recording rules or alerting rules.
====

> * Added the `PrometheusPossibleNarrowSelectors` alert to warn users when {cmo-first} detects that Prometheus query selectors might not take into account that values on the `le` label of classic histograms or the `quantile` label of summaries could also be floats.

Reviewer: ```suggestion
* Added the `PrometheusPossibleNarrowSelectors` alert to warn users when {cmo-first} detects that Prometheus query or relabel configs could be too restrictive, it takes into account that values on the `le` label of classic histograms or the `quantile` label of summaries could also be floats.
```
not sure @machine424 has better description for it

[id="ocp-4-19-monitoring-prometheus-v3-upgrade"]

> ==== Prometheus v3 upgrade

Author: I could add more information to the sentence for upstream link, something like:

> For more information about the changes between Prometheus v2 and v3, see link....

Would something like that work?

> ==== Prometheus v3 upgrade

Reviewer 2: > For more information about the other changes between Prometheus v2 and v3, see link....

works for me

> * Added the `PrometheusPossibleNarrowSelectors` alert to warn users when Prometheus queries or relabel configurations use selectors that could be too restrictive and might not take into account that values on the `le` label of classic histograms or the `quantile` label of summaries are floats in Prometheus v3. For more information, see the "Prometheus v3 upgrade" section.

Reviewer 2: ```suggestion
* Added the `PrometheusPossibleNarrowSelectors` alert to warn users when PromQL queries or metric relabel configurations use selectors that could be too restrictive and might not take into account that values on the `le` label of classic histograms or the `quantile` label of summaries are floats in Prometheus v3. For more information, see the "Prometheus v3 upgrade" section.
```

to be more precise. (thinking about this while writing the runbook)

> * Added the `PrometheusPossibleNarrowSelectors` alert to warn users when Prometheus queries or relabel configurations use selectors that could be too restrictive and might not take into account that values on the `le` label of classic histograms or the `quantile` label of summaries could also be floats. For more information, see the following "Prometheus v3 upgrade" section.

Reviewer 2: ```suggestion
* Added the `PrometheusPossibleNarrowSelectors` alert to warn users when Prometheus queries or relabel configurations use selectors that could be too restrictive and might not take into account that values on the `le` label of classic histograms or the `quantile` label of summaries could are floats in Prometheus v3. For more information, see the following "Prometheus v3 upgrade" section.
```

For more information, see link:https://prometheus.io/docs/prometheus/latest/migration/[Prometheus 3.0 migration guide].

[id="ocp-4-19-monitoring-metrics-collection-profiles-ga"]
==== Metrics collection profiles (General Availability)

{product-title} 4.13 introduced the ability to set a metrics collection profile for default platform monitoring to collect either the default amount of metrics data or a minimal amount of metrics data. In {product-title} {product-version}, metrics collection profiles are now generally available.

For more information, see xref:../observability/monitoring/about-ocp-monitoring/key-concepts.adoc#configuring-metrics-collection-profiles_key-concepts[About metrics collection profiles] and xref:../observability/monitoring/configuring-core-platform-monitoring/configuring-performance-and-scalability.adoc#choosing-a-metrics-collection-profile_configuring-performance-and-scalability[Choosing a metrics collection profile].

[id="ocp-4-19-monitoring-added-cluster-proxy-support-for-external-alertmanager-instances"]
==== Added cluster proxy support for external Alertmanager instances

> With this release, external Alertmanager instances now use the cluster-wide HTTP proxy settings for communication. The {cmo-first} reads the cluster-wide proxy settings and configures the appropriate proxy URL for the Alertmanager endpoints.

Author: approved by @marioferh through Slack

[id="ocp-4-19-monitoring-strict-validation-for-cmo-is-improved"]
==== Strict validation for the {cmo-full} is improved

> With this release, the strict validation introduced in {product-title} 4.18 is improved. Error messages now clearly identify the affected field, and validation is case-sensitive to ensure more accurate and consistent configuration. 

Reviewer 3: ```suggestion
With this release, the strict validation introduced in {product-title} 4.18 is improved. Error messages now clearly identify the affected field, and validation is case-sensitive to ensure a more accurate and consistent configuration. 
```
